core/arch/arm/sm/sm.c

```diff
@@ -31,17 +32,23 @@ uint32_t sm_from_nsec(struct sm_ctx *ctx)
 		return SM_EXIT_TO_NON_SECURE;
 #endif
 
-#ifdef CFG_PSCI_ARM32
-	if (OPTEE_SMC_OWNER_NUM(*nsec_r0) == OPTEE_SMC_OWNER_STANDARD) {
-		smc_std_handler((struct thread_smc_args *)nsec_r0, &ctx->nsec);
-		return SM_EXIT_TO_NON_SECURE;
+	if (IS_ENABLED(CFG_PSCI_ARM32)) {
```

Comment:
I don't think it right to make SMCCC v1.1 depend on PSCI. Either we always enable it, my preference, or we add another config variable.

Reply:
The Linux kernel does and probes ARM_ARCH_VERSION func-id support through the PSCI_FEATURE func-id. [`invoke_psci_fn(ARM_SMCCC_VERSION_FUNC_ID, 0, 0, 0);`](https://elixir.bootlin.com/linux/v5.6.8/source/drivers/firmware/psci/psci.c#L413)

Reply:
Actually, the spec SMCCC v1.2 specifies PSCI_FEATURES supports SMCCC_VERSION:
```
PSCI_FEATURES
Parameters uint32 Function ID    0x8400 000A
           uint32 psci_func_id   Function ID for a PSCI Function or SMCCC_VERSION